### PR TITLE
[WIP] Add test node

### PR DESCRIPTION
**Pull request by Bzil**
TODO 

### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/namespaces/nodes.yaml [hunk 1]
```diff
@@ -268,6 +268,21 @@ paths:
       responses:
         '200':
           $ref: '#/components/responses/nodes.info@200'
+  /_nodes/{metric}/stats:
```
Comment (Bzil) [inline spec/namespaces/nodes.yaml:271]: // Read https://opensearch.org/docs/latest/api-reference/nodes-apis/index/

File: tests/_core/nodes.yaml [hunk 2]
```diff
@@ -7,3 +7,53 @@ chapters:
...
+      metric:
+        - _cluster_manager
+    response:
+      status: 200
+  - synopsis: Nodes _cluster_manager metric with a json response.
+    path: /_nodes/{metric}/stats
+    method: GET
+    parameters:
+      metric:
+        - data:true
```
Comment (Bzil) [inline tests/_core/nodes.yaml:48]: @dblock do you have any idea how to implement a fix for this test ?

Comment (dblock) [reply tests/_core/nodes.yaml:48]: This error?

```
FAILED  Nodes _cluster_manager metric with a json response. 
            FAILED  PARAMETERS 
                FAILED  metric (data/0 must be equal to one of the allowed values)
```

The `data:true` value doesn't seem like a valid metric.

Comment (Bzil) [reply tests/_core/nodes.yaml:48]: it's seems weird yes, but it's define in documentation

Comment (dblock) [reply tests/_core/nodes.yaml:48]: Is it read like a JSON value? Quote it, `"data:true"`?

Comment (Bzil) [reply tests/_core/nodes.yaml:48]: if I do that, that implies that I have to put it in the enum and therefore do I have to do the same for `data:false` and the other filters?

Comment (dblock) [reply tests/_core/nodes.yaml:48]: I suppose that yes? There's really no such thing as booleans in URL paths AFAIK, this is a very creative use of the API.

File: spec/schemas/nodes.info.yaml [hunk 3]
```diff
@@ -21,21 +21,24 @@ components:
...
+                - type: object
+                  $ref: '#/components/schemas/NodeInfo'
+                - type: object
+                  additionalProperties: {}
             _nodes:
                 $ref: 'nodes._common.yaml#/components/schemas/NodesResponseBase'
           required:
             - cluster_name
-            - nodes
             - _nodes
```
Comment (Xtansia) [inline spec/schemas/nodes.info.yaml:42]: These changes are likely due to the duplicate path and having the wrong schemas, please revert them and test again with other changes made.

File: tests/_core/nodes.yaml [hunk 4]
```diff
@@ -7,3 +7,53 @@ chapters:
     method: GET
```
Comment (Xtansia) [inline tests/_core/nodes.yaml:7]: This file should be moved under `tests/nodes/` and split into `info.yaml` and `stats.yaml`

File: tests/_core/nodes.yaml (hunk 2 above)
Comment (Xtansia) [reply tests/_core/nodes.yaml:48]: These should be using `/_nodes/{node_id}/stats`

File: tests/_core/nodes.yaml [hunk 5]
```diff
@@ -7,3 +7,53 @@ chapters:
...
+    path: /_nodes/{metric}
+    method: GET
+    parameters:
+      metric:
+        - jvm
+        - thread_pool
+    response:
+      status: 200
+  - synopsis: Nodes _all metric with a json response.
+    path: /_nodes/{metric}
```
Comment (Xtansia) [inline tests/_core/nodes.yaml:28]: This one is technically `/_nodes/{node_id}`

**Review by Xtansia — CHANGES_REQUESTED** [review]

**Comment by dblock** [comment]
Last tests from this PR made it into https://github.com/opensearch-project/opensearch-api-specification/pull/455.

**Pull request closed without merging**